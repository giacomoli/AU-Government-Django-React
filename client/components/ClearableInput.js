import React from 'react'
import PropTypes from 'prop-types'

const ClearableInput = ({ onClear, onChange, value, ...props }) => (
  <div style={{ position: 'relative' }}>
    <input value={value} onChange={onChange} type="text" {...props} />
    <i onClick={onClear} tabIndex="-1" role="button" className="fa fa-times clear" />
    <style jsx>{`
      .clear {
        display: ${value ? 'inline' : 'none'};
        position: absolute;
        right: 15px;
        top: 10px;
        cursor: pointer;
      }
    `}</style>
  </div>
)

ClearableInput.propTypes = {
  onClear: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
}

ClearableInput.defaultProps = {
  value: '',
}

export default ClearableInput
