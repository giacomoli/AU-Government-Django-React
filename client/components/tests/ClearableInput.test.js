import React from 'react'
import renderer from 'react-test-renderer'
import { ApolloProvider } from 'react-apollo';
import { reducer } from 'redux-form'
import { ApolloClient } from 'apollo-client';
import { createStore } from 'redux';
import ClearableInput from '../ClearableInput'

/* eslint-disable */
describe('ClearableInput', () => {
  const client = new ApolloClient()
  const store = createStore(reducer)
  const onChange = jest.fn()
  const onClear = jest.fn()

  it('should render ClearableInput', () => {
    const wrapper = renderer.create(
      <ApolloProvider client={client} store={store}>
        <ClearableInput
          onClear={onClear}
          onChange={onChange}
          value="qwe"
        />
      </ApolloProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render ClearableInput without value', () => {
    const wrapper = renderer.create(
      <ApolloProvider client={client} store={store}>
        <ClearableInput
          onClear={onClear}
          onChange={onChange}
          />
      </ApolloProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
