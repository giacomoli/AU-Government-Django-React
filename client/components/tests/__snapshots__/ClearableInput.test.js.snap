// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearableInput should render ClearableInput 1`] = `
<div
  className="jsx-2047498976"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <input
    className="jsx-2047498976 "
    onChange={[MockFunction]}
    type="text"
    value="qwe"
  />
  <i
    className="jsx-2047498976 fa fa-times clear"
    onClick={[MockFunction]}
    role="button"
    tabIndex="-1"
  />
</div>
`;

exports[`ClearableInput should render ClearableInput without value 1`] = `
<div
  className="jsx-1157770627"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <input
    className="jsx-1157770627 "
    onChange={[MockFunction]}
    type="text"
    value=""
  />
  <i
    className="jsx-1157770627 fa fa-times clear"
    onClick={[MockFunction]}
    role="button"
    tabIndex="-1"
  />
</div>
`;
